.github/workflows/upload-build-to-s3.yaml

```diff
@@ -62,48 +66,220 @@ jobs:
           make clean
           make download-licenses 
           make FINCH_OS_IMAGE_LOCATION_ROOT=/Applications/Finch
-          tar -zcvf finch.${GITHUB_REF_NAME}.x86_64.tar.gz _output
         shell: zsh {0}
 
-      - name: Upload macos x86_64 build
-        uses: actions/upload-artifact@v2
+      - name: configure aws credentials
+        uses: aws-actions/configure-aws-credentials@v2
         with:
-          name: finch.macos-x86_64
-          path: finch.*.x86_64.tar.gz
-          if-no-files-found: error
+          role-to-assume: ${{ secrets.ROLE }}
+          role-session-name: dependency-upload-session
+          aws-region: ${{ secrets.REGION }}
+
+      - name: generate pkg
+        run: |
+          ./installer-builder/tools/release-installer.sh x86_64 ${GITHUB_REF_NAME} ${{ secrets.INSTALLER_PRIVATE_BUCKET_NAME }} ${{ secrets.EXECUTABLE_BUCKET }} ${{ secrets.PKG_BUCKET }} ${{ secrets.NOTARIZATION_ACCOUNT }} ${{ secrets.NOTARIZATION_CREDENTIAL }}
+        shell: zsh {0}
 
-  release:
-    runs-on: ubuntu-latest
-    timeout-minutes: 5
+  macos-aarch64-pkg-test:
```

Comment:
can we add a new matrix for `architecture` set to `aarch64` and `x86_64 ` and remove the duplication here? Seems like macos-aarch64-pkg-test and macos-x86-64-pkg-test are almost exactly the same.

Reply:
Yes, I can make that with a following up PR to avoid regression. Current code is merged from the existing action for installer test